Initial support for user roles

jonc125 opened the pull request:
Part of #34, this adds user role info to the DB, with a simple management interface (only visible to 'admin' users), and restricts host operations to users with the 'owner' role.

jonc125 committed:
Add Roles table, approach 1

This has a many-many relationship with User. The advantage is that
adding new roles is a DB content change, rather than a code change.
The disadvantge is that code needs to check role names against
strings to determine user permissions.

jonc125 committed:
A little tidying

jonc125 committed:
Use a Python Enum to define available roles

jonc125 committed:
Add a view decorator to check the user's role

Also
- the test user automatically becomes an admin
- only users with the 'owner' role can add/modify hosts

jonc125 committed:
Add a simple admin interface to manage user roles

jonc125 commented:
The travis failure is PEP8:
```
./src/cloudlabs/models.py:27:80: E501 line too long (99 > 79 characters)
./src/cloudlabs/admin/views.py:49:80: E501 line too long (85 > 79 characters)
./src/cloudlabs/admin/views.py:51:80: E501 line too long (89 > 79 characters)
```

I have the max line length set to 100 locally. Do you have a preference as to what we use for this project @raquel-ucl?

raquelalegre commented:
I just left the default which is 79. Do change it to 100 if that'll make your life easier. You probably know, but you can by adding this line to tox.ini file:

```
max-line-length = 100
```

jonc125 committed:
Increase line length allowed for PEP8 check

jonc125 committed:
Remove debugging lines

jonc125 committed:
Add a bit more description for dns_name

jonc125 committed:
Fix for Python 3.5

jonc125 committed:
Do a `terraform init` before each deploy

This should make it work on machines that haven't run Terraform before,
but isn't an ideal solution, since it downloads the Azure provider each
time.

jonc125 committed:
Don't automatically upgrade the Azure provider

jonc125 committed:
Add some Terraform info to the README

jonc125 committed:
Allow our internal deployment SSH key to login as well

jonc125 committed:
Upgrade the Terraform used by Travis

It seems the 'version' key to specify Azure provider version is a
new feature!
